### PR TITLE
[mqtt][homie] Use UID-encoder/decoder for channel/device-mapping

### PR DESCRIPTION
**Pull request by J-N-K**

### CODE REVIEW

File: bundles/org.openhab.binding.mqtt.homie/src/main/java/org/openhab/binding/mqtt/homie/internal/homie300/Property.java [hunk 1]
```diff
@@ -83,7 +84,7 @@ public Property(String topic, Node node, String propertyID, DeviceCallback callb
         this.topic = topic + "/" + propertyID;
         this.parentNode = node;
         this.propertyID = propertyID;
-        channelUID = new ChannelUID(node.uid(), propertyID);
+        channelUID = new ChannelUID(node.uid(), UIDUtils.encode(propertyID));
         channelTypeUID = new ChannelTypeUID(MqttBindingConstants.BINDING_ID, this.topic.replace('/', '_'));
```
Comment (davidgraeff) [inline bundles/org.openhab.binding.mqtt.homie/src/main/java/org/openhab/binding/mqtt/homie/internal/homie300/Property.java]: Encoding is required here as well I think

**Comment by J-N-K** [comment]
@davidgraeff I can't find the issue where this was found, can you remember which one it was?

**Comment by martinvw** [comment]
> Is think this is a breaking change and requires mentioning in the distro release notes document, based on that existing channels are rendered unused and those renamed ones will pop on in the inbox.

Feel free to merge when you think that all others tasks are completed :-)

**Comment by davidgraeff** [comment]
No don't have the issue at hand

**Pull request merged**